### ISSUE TITLE
cmdserver is started and killed every operation during normal use

### ISSUE THREAD

**Issue by ncalexan** [ananthakumaran/monky#109]
I've used monky for ages and just noticed that during "normal use" the cmdserver process will be started and killed every operation.  At least, that's what I witness and what the use of `unwind-protect` [here](https://github.com/ananthakumaran/monky/blob/e04632277ef24acacc029ae29db1fadc458ae83b/monky.el#L605).

The user can work around this with a manual `(setq monky-cmd-process (monky-cmdserver-start))` but that's both awkward and not documented.

Perhaps this was supposed to be `condition-case` so that the `delete-process` only happens in error cases?  Or perhaps there's supposed to be some other flow in normal use that I don't know about?

**Comment by Wilfred** [comment]
Yeah, I agree that looks broken. I've gradually been trying to reduce the amount of macros in monky, because I find the code hard to reason about.
